### PR TITLE
 Dive in Gorm setup

### PR DESCRIPTION
**Pull request by BarnabePILLIAUDIN**
# WAITING FOR PROD DB TO BE SETUP

# GORM dive in 

We will be using GORM as ORM, query builder and migration engine also via it automigrate. As we wasn't able to make atlas work properly and automigrate should be enough for our project

### CODE REVIEW

File: docs/dive-in/2024/02/2024-02-07:configure-ent.md [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+Date : 2024-02-07
+
+## SUMMARY:
+We want to declare a schema of our database and this schema to be applied to the database.
+But we don't want it to be like dark magic. We want to have some control over it.
+As we are using golang for our backend we think about using gorm and atlas to achieve that.
+
+## SOLUTION:
+
+Atlas offers a clean to do versioned migrations.
```
Comment (Magali-D) [inline docs/dive-in/2024/02/2024-02-07:configure-ent.md]: Can you explain?

File: docs/dive-in/2024/02/2024-02-07:configure-ent.md [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+```shell 
+atlas schema clean -u "dburi"
+```
+
+### Solution 
+
+After some discutions with a teacher and go developers it seems that's gorm automigrate would be 
+enough for our project. If needed we can still create SQL script to correct things that automigrate would have done wrong.
+It has the advantage that the migrations will be applied each time that the api is started so we won't need to modify 
+the docker fiel
```
Comment (Magali-D) [inline docs/dive-in/2024/02/2024-02-07:configure-ent.md]: Do you mean Dockerfile?

**Review by Magali-D — COMMENTED** [review]
Dive-in is unclear concerning Atlas usage. It is still often presented are our solution (ex: "To make the migrations we will use Atlas.") even though the conclusion states otherwise.

File: docs/dive-in/2024/02/2024-02-07:configure-ent.md (hunk 1 above)
Comment (barnabepilliaudin) [reply docs/dive-in/2024/02/2024-02-07:configure-ent.md]: Atlas offers a way to have versionned migrations, which means migrations are in files that are versionned in the repo. It's different that automatic migrations that are run at the start of the api and doesn't generate versionned file.

File: docs/dive-in/2024/02/2024-02-07:configure-ent.md (hunk 2 above)
Comment (barnabepilliaudin) [reply docs/dive-in/2024/02/2024-02-07:configure-ent.md]:
```suggestion
the Dockerfile
```

**Comment by BarnabePILLIAUDIN** [comment]
> Dive-in is unclear concerning Atlas usage. It is still often presented are our solution (ex: "To make the migrations we will use Atlas.") even though the conclusion states otherwise.

@Magali-D  Edited this part but I don't want to completely remove Atlas from the dive in as it was a big part of the research and it would interesting to keep a track of it

File: docs/dive-in/2024/02/2024-02-07:configure-ent.md [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+
+type Car struct {
+gorm.Model
+Owner       User   `json:"owner" gorm:"-"`
+Constructor string `json:"constructor"`
+ModelName   string `json:"modelName"`
+}
+```
+
+To make the migrations could use Atlas.It offers a clean to do versioned migrations.
```
Comment (Magali-D) [inline docs/dive-in/2024/02/2024-02-07:configure-ent.md]:
```suggestion
To make the migrations we could use Atlas. It offers a clean way to do versioned migrations.
```

File: docs/dive-in/2024/02/2024-02-07:configure-ent.md [hunk 4]
```diff
@@ -0,0 +1,121 @@
+# Dive-in: Configure gorm 
+
+Author : Barnabé PILLIAUDIN  
+Co-Assignee: Cédric Gautier  
+Date : 2024-02-07
+
+## SUMMARY:
+We want to declare a schema of our database and this schema to be applied to the database.
+But we don't want it to be like dark magic. We want to have some control over it.
+As we are using golang for our backend we think about using gorm and atlas to achieve that.
```
Comment (cedricgautier) [inline docs/dive-in/2024/02/2024-02-07:configure-ent.md]:
```suggestion
We want to declare a schema of our data models that will be applied to our database.
But we don't want it to be like dark magic. We want to have some control over it.
As we are using Golang for our backend we are thinking about using GORM and Atlas to achieve that.
```

File: docs/dive-in/2024/02/2024-02-07:configure-gorm.md [hunk 5]
```diff
@@ -0,0 +1,139 @@
...
+But we don't want it to be like dark magic. We want to have some control over it.
+As we are using Golang as our backend language, we'll need to provide an ORM for our language.
+
+## RESEARCH AND TESTS:
+
+What we're looking for :
+
+- A migration system
+- A ORM to handle the logic required to interact with databases.
+- A query builder that will build SQL statements from defining our models our backend langauge.
```
Comment (Magali-D) [inline docs/dive-in/2024/02/2024-02-07:configure-gorm.md]:
```suggestion
- A query builder that will build SQL statements from defining our models our backend language.
```
Is a word missing as well?

**Review by cedricgautier — APPROVED** [review]

**Review by cedricgautier — APPROVED** [review]

**Review by Magali-D — APPROVED** [review]
LGTM

**Pull request merged**